admin/registration_providers/views.py

```diff
@@ -0,0 +1,375 @@
+        kwargs['show_taxonomies'] = True
+        kwargs['tinymce_apikey'] = settings.TINYMCE_APIKEY
+        return super(CreateRegistrationProvider, self).get_context_data(*args, **kwargs)
+
+
+class RegistrationProviderList(PermissionRequiredMixin, ListView):
+    paginate_by = 25
+    template_name = 'registration_providers/list.html'
+    ordering = 'name'
+    permission_required = 'osf.view_registrationprovider'
+    raise_exception = True
+    model = RegistrationProvider
+
+    def get_queryset(self):
+        return RegistrationProvider.objects.all().order_by(self.ordering)
+
+    def get_context_data(self, **kwargs):
+        query_set = kwargs.pop('object_list', self.object_list)
+        page_size = self.get_paginate_by(query_set)
+        paginator, page, query_set, is_paginated = self.paginate_queryset(
+            query_set, page_size)
+        return {
+            'registration_providers': query_set,
+            'page': page,
+        }
+
+
+class RegistrationProviderDetail(PermissionRequiredMixin, View):
+    permission_required = 'osf.change_registrationprovider'
+    raise_exception = True
+
+    def get(self, request, *args, **kwargs):
+        view = RegistrationProviderDisplay.as_view()
+        return view(request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        view = RegistrationProviderChangeForm.as_view()
+        return view(request, *args, **kwargs)
+
+
+class RegistrationProviderDisplay(PermissionRequiredMixin, DetailView):
+    model = RegistrationProvider
+    template_name = 'registration_providers/detail.html'
+    permission_required = 'osf.view_registrationprovider'
+    raise_exception = True
+
+    def get_object(self, queryset=None):
+        return RegistrationProvider.objects.get(id=self.kwargs.get('registration_provider_id'))
+
+    def get_context_data(self, *args, **kwargs):
+        registration_provider = self.get_object()
+        registration_provider_attributes = model_to_dict(registration_provider)
+        registration_provider_attributes['default_license'] = registration_provider.default_license.name if registration_provider.default_license else None
+
+        # compile html list of licenses_acceptable so we can render them as a list
+        licenses_acceptable = list(registration_provider.licenses_acceptable.values_list('name', flat=True))
+        licenses_html = '<ul>'
+        for license in licenses_acceptable:
+            licenses_html += '<li>{}</li>'.format(license)
+        licenses_html += '</ul>'
+        registration_provider_attributes['licenses_acceptable'] = licenses_html
+
+        # compile html list of subjects
+        subject_ids = registration_provider.all_subjects.values_list('id', flat=True)
+        kwargs['registration_provider'] = registration_provider_attributes
+        kwargs['subject_ids'] = list(subject_ids)
+
+        subject_html = '<ul class="subjects-list">'
+        for parent in registration_provider.top_level_subjects:
```

Comment:
There's also some weirdness in here -- `top_level_subjects` will return bepress if the provider has none (this was done for CollectionProviders, which don't currently have the custom creation form). There isn't a parent check similar to `child.id in subject_ids` happening, leading to subjects that aren't actually connected being displayed.